src/tests/fs_tests.rs

```diff
@@ -4066,4 +4067,136 @@ pub mod fs_tests {
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        let mut statdata = StatData::default();
+
+        // test out whether an error is output for a non existent file path
+        // (ENOENT[-2])
+        assert_eq!(
+            cage.stat_syscall("non_existent_file_path", &mut statdata),
+            syscall_error(Errno::ENOENT, "stat", "test_failure")
+        );
+
+        // setting up directory inode object '/tmp' for testing stat_syscall with a
+        // directory
+        let dir_path = "/tmp"; // since setup already initializes tmp, assuming it is there
+        assert_eq!(cage.stat_syscall(dir_path, &mut statdata), 0);
+
+        // setting up generic inode object "/tmp/generic" for testing stat_syscall with
+        // a generic file
+        let generic_path = "/tmp/generic";
+        let creat_fd = cage.creat_syscall(generic_path, S_IRWXA);
+        assert!(creat_fd > 0);
+        assert_eq!(cage.stat_syscall(generic_path, &mut statdata), 0);
+
+        // setting up character device inode object "/chardev" for testing stat_syscall
+        // with a character device
+        let dev = makedev(&DevNo { major: 1, minor: 3 });
+        let chardev_path = "/chardev";
+        assert_eq!(
+            cage.mknod_syscall(chardev_path, S_IRWXA | S_IFCHR as u32, dev),
+            0
+        );
+        assert_eq!(cage.stat_syscall(chardev_path, &mut statdata), 0);
+
+        // setting up socket inode object with path "/socket.sock"  for testing
+        // stat_syscall with a socket
+        let socketfile_path = "/socket.sock";
+        let socketfd = cage.socket_syscall(AF_UNIX, SOCK_STREAM, 0);
+        assert!(socketfd > 0);
+        let sockaddr = interface::new_sockaddr_unix(AF_UNIX as u16, socketfile_path.as_bytes());
+        let socket = interface::GenSockaddr::Unix(sockaddr);
+        assert_eq!(cage.bind_syscall(socketfd, &socket), 0);
+
+        // stat_syscall test here
+        assert_eq!(cage.stat_syscall(socketfile_path, &mut statdata), 0);
+
+        // socket teardown
+        assert_eq!(cage.close_syscall(socketfd), 0);
+        cage.unlink_syscall(socketfile_path);
+
+        lindrustfinalize();
+        return;
+    }
+
+    pub fn ut_lind_fs_fstat_syscall_tests() {
+        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        let mut statdata = StatData::default();
+
+        // test whether an invalid fd results in a panic
+        let invalid_fd = 0x10000000; // some dummy high fd which should fail fd check
+
+        // here, we catch the panic spat out by fstat into an Err
+        let invalid_fd_panic = catch_unwind(AssertUnwindSafe(|| {
```

Comment:
what do you think about using a #[should_panic] construct? I think it is helpful since you can also specify the expected panic message, i,e, #[should_panic(expected = "Expected panic message")]

Reply:
If I had to use should_panic, I would need to create a whole new test function for this. Using the above code, I'm able to capture the panic as an error and also perform a comparison of the error value if needed all within the same test